src/T2DMap.cpp

```diff
@@ -610,244 +586,228 @@ void T2DMap::paintEvent( QPaintEvent * e )
+    if (!pPlayerRoom) {
+        p.drawText(_w / 2, _h / 2, "No map or no valid position.");
 
         return;
     }
 
     int ox, oy; // N/U: oz;
-    if( mRID != mpMap->mRoomIdHash.value( mpHost->getName() ) && mShiftMode )
-    {
+    if (mRID != mpMap->mRoomIdHash.value(mpHost->getName()) && mShiftMode) {
         mShiftMode = false;
     }
-    TArea * pAID;
-    TRoom * pRID;
+    TArea* pAID;
+    TRoom* pRID;
     int playerArea = pPlayerRoom->getArea();
-    if( (! __Pick && ! mShiftMode ) || mpMap->mNewMove )
-    {
-
+    if ((!__Pick && !mShiftMode) || mpMap->mNewMove) {
         mShiftMode = true;
-        mpMap->mNewMove = false; // das ist nur hier von Interesse, weil es nur hier einen map editor gibt -> map wird unter Umstaenden nicht geupdated, deshalb force ich mit mNewRoom ein map update bei centerview()
+        mpMap->mNewMove =
+                false; // das ist nur hier von Interesse, weil es nur hier einen map editor gibt -> map wird unter Umstaenden nicht geupdated, deshalb force ich mit mNewRoom ein map update bei centerview()
 
-        if( !mpMap->mpRoomDB->getArea( playerArea ) )
-        {
+        if (!mpMap->mpRoomDB->getArea(playerArea)) {
             return;
         }
-        mRID = mpMap->mRoomIdHash.value( mpHost->getName() );
-        pRID = mpMap->mpRoomDB->getRoom( mRID );
+        mRID = mpMap->mRoomIdHash.value(mpHost->getName());
+        pRID = mpMap->mpRoomDB->getRoom(mRID);
         mAID = pRID->getArea();
-        pAID = mpMap->mpRoomDB->getArea( mAID );
-        if( !pRID || !pAID ) return;
+        pAID = mpMap->mpRoomDB->getArea(mAID);
+        if (!pRID || !pAID) {
+            return;
+        }
         ox = pRID->x;
-        oy = pRID->y*-1;
+        oy = pRID->y * -1;
         mOx = ox;
         mOy = oy;
         mOz = pRID->z;
-    }
-    else
-    {
-        pRID = mpMap->mpRoomDB->getRoom( mRID );
-        pAID = mpMap->mpRoomDB->getArea( mAID );
-        if( !pRID || !pAID ) return;
+    } else {
+        pRID = mpMap->mpRoomDB->getRoom(mRID);
+        pAID = mpMap->mpRoomDB->getArea(mAID);
+        if (!pRID || !pAID) {
+            return;
+        }
         ox = mOx;
         oy = mOy;
     }
-    if( ox*tx > xspan/2*tx )
-        _rx = -(tx*ox-xspan/2*tx);
-    else
-        _rx = xspan/2*tx-tx*ox;
-    if( oy*ty > yspan/2*ty )
-        _ry = -(ty*oy-yspan/2*ty);
-    else
-        _ry = yspan/2*ty-ty*oy;
+    if (ox * tx > xspan / 2 * tx) {
+        _rx = -(tx * ox - xspan / 2 * tx);
+    } else {
+        _rx = xspan / 2 * tx - tx * ox;
+    }
+    if (oy * ty > yspan / 2 * ty) {
+        _ry = -(ty * oy - yspan / 2 * ty);
+    } else {
+        _ry = yspan / 2 * ty - ty * oy;
+    }
 
-    px = ox*tx+_rx;
-    py = oy*ty+_ry;
+    px = ox * tx + _rx;
+    py = oy * ty + _ry;
 
-    TArea * pArea = pAID;
+    TArea* pArea = pAID;
 
     int zEbene;
     zEbene = mOz;
 
-    float wegBreite = 1/eSize * tx * rSize;
```

Comment:
Hmm I'll leave it to the German speaker to translate the stuff when they get about to it.